### PR TITLE
sched/env: Fix the return value of unsetenv()

### PR DESCRIPTION
**Pull request by pussuw**
If the environment variable does not exist, the function succeeds, as
defined by POSIX.

>    The unsetenv() function deletes the variable name from the
       environment.  If name does not exist in the environment, then the
       function succeeds, and the environment is unchanged.

## Summary
Fixes return value of unsetenv, value was accidentally broken in #6083

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: sched/environ/env_unsetenv.c [hunk 1]
```diff
@@ -61,23 +61,22 @@ int unsetenv(FAR const char *name)
...
       /* It does!  Remove the name=value pair from the environment. */
 
-      env_removevar(group, ret);
-      ret = OK;
+      env_removevar(group, idx);
     }
 
   sched_unlock();
-  return ret;
+  return OK;
```
Comment (pkarashchenko) [inline sched/environ/env_unsetenv.c:79]: According to POSIX `unsetenv` can fail with
```
[EINVAL]
    The name argument points to an empty string, or points to a string containing an '=' character. 
```
Leaving here just for a note. Should be fixed by a separate PR.

**Review by pkarashchenko — APPROVED** [review]

File: sched/environ/env_unsetenv.c (hunk 1 above)
Comment (pussuw) [reply sched/environ/env_unsetenv.c:79]: Yes we could add check for the incoming parameter too. The man page is a bit ambiguous in my opinion about this.

For `setenv()`it makes perfect sense to check for the incoming parameter. 
For `unsetenv()` it is ambiguous. The check is for parameter validity, but no such env variable can exist, as setenv() checks for parameter validity. As such name does not exist, should the function return success ?

Comment (pussuw) [reply sched/environ/env_unsetenv.c:79]: Yup, there has clearly been some iteration with that since I can find a contradicting (at least ambiguous!) man page (first result in google): https://man7.org/linux/man-pages/man3/setenv.3.html

But like said, neither **prohibit** returning EINVAL if the incoming parameter "name" is not valid so it is better to do the test.

Comment (pussuw) [reply sched/environ/env_unsetenv.c:87]: Added the parameter check as separate commit

File: sched/environ/env_unsetenv.c [hunk 2]
```diff
@@ -61,23 +61,30 @@ int unsetenv(FAR const char *name)
...
   FAR struct task_group_s *group = rtcb->group;
-  int ret = OK;
+  int idx;
 
-  DEBUGASSERT(name && group);
+  DEBUGASSERT(group);
+
+  /* Check the incoming parameter */
+
+  if (name == NULL || *name == '\0' || strchr (name, '=') != NULL)
```
Comment (pkarashchenko) [inline sched/environ/env_unsetenv.c:70]:
```suggestion
  DEBUGASSERT(name && group);

  /* Check the incoming parameter */

  if (*name == '\0' || strchr (name, '=') != NULL)
```

File: sched/environ/env_unsetenv.c [hunk 3]
```diff
@@ -61,23 +61,30 @@ int unsetenv(FAR const char *name)
...
+    {
+      set_errno(EINVAL);
+      return -EINVAL;
```
Comment (pkarashchenko) [inline sched/environ/env_unsetenv.c:73]:
```suggestion
      return ERROR;
```

Comment (pussuw) [reply sched/environ/env_unsetenv.c:73]: Yes should return -1, will fix

File: sched/environ/env_unsetenv.c (hunk 2 above)
Comment (pussuw) [reply sched/environ/env_unsetenv.c:70]: My thought was not to crash the whole system here but I can revert to the old implementation too.

Comment (pkarashchenko) [reply sched/environ/env_unsetenv.c:70]: I'm not sure. We can leave `NULL` check as `EINVAL`. That makes sense. I just was thinking that POSIX states nothing about NULL case, but only covers empty string and `'='` holder string. I'm fine to keep the `NULL` check as recoverable error. Not sure how Linux manages that.

Comment (pussuw) [reply sched/environ/env_unsetenv.c:70]: Yes glibc tests for the NULL pointer

Comment (gustavonihei) [inline sched/environ/env_unsetenv.c:70]:
```suggestion
  if (name == NULL || *name == '\0' || strchr(name, '=') != NULL)
```
I believe that white space should not exist.

Comment (gustavonihei) [inline sched/environ/env_unsetenv.c:70]:
```suggestion
  if (name == NULL || name[0] == '\0' || strchr (name, '=') != NULL)
```
nit: just a suggestion, at least for me this way becomes more easily readable.

Comment (pussuw) [reply sched/environ/env_unsetenv.c:70]: At least setenv() uses the original style `*name == '\0'`also so I'll keep the original formatting

Comment (pussuw) [reply sched/environ/env_unsetenv.c:70]: Thanks, I missed this

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]
LGTM
